includes/class-admin.php

```diff
@@ -758,4 +761,47 @@ public static function dashboard_glance_items( $items ) {
 
 		return $items;
 	}
+
+	/**
+	 * Sanatize the identifier
+	 *
+	 * @param string $id The identifier.
+	 *
+	 * @return false|string The sanatized identifier or false if it is already in use.
+	 */
+	private static function sanatize_identifier( $id ) {
+		if ( empty( $id ) ) {
+			return false;
+		}
+
+		// hack to allow dots in the username
+		$parts     = explode( '.', $id );
+		$sanitized = array();
+
+		foreach ( $parts as $part ) {
+			$sanitized[] = \sanitize_title( $part );
+		}
+
+		$sanitized = implode( '.', $sanitized );
+
+		// check for login or nicename.
+		$user = new WP_User_Query(
+			array(
+				'search'         => $sanitized,
+				'search_columns' => array( 'user_login', 'user_nicename' ),
+				'number'         => 1,
+				'hide_empty'     => true,
+				'fields'         => 'ID',
+			)
+		);
+
+		if ( $user->get_results() ) {
+			return new \WP_Error(
+				'identifier_exists',
+				\__( 'This identifier is already in use.', 'activitypub' )
+			);
+		}
```

Comment:
This doesn't check whether the same identifier was already set for another username.